src/mon/OSDMonitor.cc

```diff
@@ -3552,6 +3552,21 @@ bool OSDMonitor::preprocess_boot(MonOpRequestRef op)
     goto ignore;
   }
 
+  // See crimson/osd/osd.cc: OSD::_send_boot
+  if (m->metadata.contains("osd_type")) {
+    // m->metadata["osd_type"] must be "crimson", classic doesn't send osd_type
```

Comment:
assert?

Reply:
No, I don't want to crash the mon.

Reply:
Fair point, fixed to output a message to derr but to otherwise allow it.